src/charm.py

```diff
@@ -516,7 +522,7 @@ def bootstrap_cluster(self, event: ops.framework.EventBase) -> None:
 
             if error_socket_not_exists in e.stderr:
                 event.defer()
-                return
+                raise sunbeam_guard.WaitingExceptionError("waiting for snap")
 
             if error_already_exists not in e.stderr:
                 raise e
```

Comment:
Suggest to add info logging here (on raise)
And debug logging at normal method exit

Reply:
success case log added at line 522